crates/ruma-client-api/src/error.rs

```diff
@@ -277,4 +303,18 @@ mod tests {
             }
         );
     }
+    #[test]
+    fn convert_from_formhttprequest_error() {
+        let deserialization_result: Result<Value, serde_json::Error> =
+            serde_json::from_str(r#"notajsonbutarandomstring"#);
+        match deserialization_result {
+            Ok(_) => assert_eq!(1, 2), //intentional fail to notify of changed serde API if ever.
```

Comment:
```suggestion
            Ok(_) => unreachable!(" serde has changed it's API"),
```

Reply:
Just `unreachable!()` should be enough ^^